src/libkomwm.py

```
@@ -267,6 +267,104 @@ def print_warning(msg):
             idx = unique_prios.index(prio_ranges[prio_range]['priorities'][prio_id])
             prio_ranges[prio_range]['priorities'][prio_id] = int(step * (base_idx + idx))
 
+def load_colors(file_path:str) -> set[int]:
+    colors = set()
+    if os.path.exists(file_path):
+        with open(file_path, "rt") as colors_in_file:
+            for colorLine in colors_in_file:
+                if colorLine.strip():
+                    colors.add(int(colorLine.strip()))
+    return colors
+
+def save_colors(colors:set[int], file_path:str):
+    with open(file_path, "wt") as colors_file:
+        for c in sorted(colors):
+            colors_file.write("%d\n" % c)
+
+def addPattern(patterns:list[list[float]], dashes: list[float]):
+    if dashes and dashes not in patterns:
+        patterns.append(dashes)
+
+def load_patterns(patterns_file_name:str) -> list[list[float]]:
+    patterns = []
+
+    if os.path.exists(patterns_file_name):
```

Should it throw on missing file?

---

Funny thing: if you delete `data/patterns.txt` file from OrganicMaps repo then `tools/unix/generate_drules.sh` will regenerate the content from *.mapcss rules.

This happens here: [libkomwm.py:763](https://github.com/organicmaps/kothic/blob/kothic-improvements/src/libkomwm.py#L763), [libkomwm.py:791](https://github.com/organicmaps/kothic/blob/kothic-improvements/src/libkomwm.py#L791)

---

"patterns.txt" is the auto generated file like drules_proto_*.txt/bin